app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -35,7 +38,9 @@ def failure
 
   private
 
-  # TODO: [thepracticaldev/oss] test all of this
+  # TODO: [thepracticaldev/oss] investigate if there's a way to simplify this
+  # by using Devise+OmniAuth builtin integration
+  # <https://github.com/heartcombo/devise/wiki/OmniAuth:-Overview>
```

Comment:
OmniAuth has builtin support for Devise, I wonder if there's a way to have less code or not, I'll investigate post refactoring as there's no need to slow down the extraction of providers for this